universal-application-tool-0.0.1/app/models/Applicant.java

```diff
@@ -16,7 +16,17 @@
 import play.data.validation.Constraints;
 import services.applicant.ApplicantData;
 
-/** The ebean mapped class that represents an individual applicant */
+/**
+ * An EBean mapped class that represents a single applicant at a specific version.
+ *
+ * <p>This class primarily exists to handle database interaction for its {@code object} property
+ * which it uses to hydrate a {@code ApplicantData} instance. See {@code
+ * Applicant#getApplicantData}.
+ *
+ * <p>Note that an {@code Applicant} is not one-to-one with an actual human applicant: {@code
```

Comment:
Thoughts: I think it is common that an account associates with only one applicant. Many to one mapping only happens when an user tries to merge accounts (turning from a guest account to a IDCS account): you get two accounts, one for each sign-in method, and then the two applicant data are merged and associate with the new (IDCS) account.

Reply:
Many to one also happens when an applicant moves on to a new version of the system that may have different programs available and question definitions.

Reply:
Hung-Ming is right I believe.

Reply:
updated, please TAL